tags/reserved_tags.go

```diff
@@ -0,0 +1,118 @@
+const XAkitaCI Key = "x-akita-ci"
+
+// == CircleCI tags ===========================================================
+
+// The contents of the CIRCLE_BUILD_URL environment variable. Attached to
+// traces and specs derived from a CircleCI job.
+const XAkitaCircleCIBuildURL Key = "x-akita-circleci-build-url"
+
+// == Travis tags =============================================================
+
+// The contents of the TRAVIS_BUILD_WEB_URL environment variable. Attached to
+// traces and specs derived from a Travis job.
+const XAkitaTravisBuildWebURL Key = "x-akita-travis-build-web-url"
+
+// The contents of the TRAVIS_JOB_WEB_URL environment variable. Attached to
+// traces and specs derived from a Travis job.
+const XAkitaTravisJobWebURL Key = "x-akita-travis-job-web-url"
+
+// == Generic git tags ========================================================
+
+// Identifies the git branch from which the trace or spec was derived. Attached
+// to traces or specs obtained from CI.
+const XAkitaGitBranch Key = "x-akita-git-branch"
+
+// Identifies the git commit hash from which the trace or spec was derived.
+// Attached to traces or specs obtained from CI.
+const XAkitaGitCommit Key = "x-akita-git-commit"
+
+// A link to the git repository. Attached to traces or specs obtained from a
+// pull/merge request.
+const XAkitaGitRepoURL Key = "x-akita-git-repo-url"
+
+// == GitHub tags =============================================================
+
+// Identifies the GitHub organization in which the pull request was made.
+// Attached to traces or specs obtained from a GitHub pull request.
+const AkitaGitHubOrganizationID Key = "x-akita-github-organization-id"
+
+// Identifies the GitHub PR number associated with the pull request. Attached
+// to traces or specs obtained from a GitHub pull request.
+const XAkitaGitHubPR Key = "x-akita-github-pr"
+
+// A link to the GitHub pull request. Attached to traces or specs obtained
+// from a GitHub pull request.
+const XAkitaGitHubPRURL Key = "x-akita-github-pr-url"
+
+// Identifies the GitHub repository for which the pull request was made.
+// Attached to traces or specs obtained from a GitHub pull request.
+const XAkitaGitHubRepo Key = "x-akita-github-repo"
+
+// == GitLab tags =============================================================
+
+const XAkitaGitLabProject Key = "x-akita-gitlab-project"
+const XAkitaGitLabMRIID Key = "x-akita-gitlab-mr-iid"
+
+// == Methods =================================================================
+
+// Determines whether a key is reserved for Akita internal use.
+func (k Key) IsReserved() bool {
+	s := strings.ToLower(string(k))
+	return strings.HasPrefix(s, "x-akita-")
+}
+
+// Returns an error if the key is reserved for Akita internal use.
+func (k Key) CheckReserved() error {
+	if !k.IsReserved() {
+		return nil
+	}
+
+	return errors.New(`Tags starting with "x-akita-" are reserved for Akita internal use.`)
+}
+
+// Returns a map from parsing a list of "key=value" pairs.
```

Comment:
Does this mean we only support one value per tag right now?  (It's not necessary to add multi-valued tags if we don't have them, but I don't know what the current tag implementation supports.)

Reply:
This came from here: https://github.com/akitasoftware/akita-cli/blob/d68d57516102f4fef68f814eadf68d4aed08be7a/cmd/internal/tags/tags.go